### DISCUSSION TITLE
Improve view mixin

### DISCUSSION THREAD

**Discussion by james-emerton** [carltongibson/django-filter#1409] [Ideas]
The current implementation of `FilterMixin` (undocumented) is not currently usable in my own class based views without a fair bit of boilerplate. I took a stab at improving it, which I will happily reshape into a PR if there is interest.

  * These changes may not be backward compatible, so the new mixin is called `FilterViewMixin`. This also makes it clear that this isn't a mixin for the `Filter` class.
  * The `filter` context variable is renamed to `filterset` for consistency
  * `get_filterset()` no longer requires the `filterset_class` attribute
  * The filterset is stored as an attribute the first time it's created and can be reused throughout the request
  * Added `get_filtered_queryset()` to retrieve the filtered queryset
  * Replace `object_list` with the filtered queryset in `get_context_data()`
  * `get_table_data()` is implemented for compatibility with django-tables2 `SingleTableMixin`

I've posted my current version as a gist: https://gist.github.com/james-emerton/8c7f23cdbe472bd6c1098692cd93436a

**Comment by carltongibson** [comment]
Hi @james-emerton — thanks for the suggestion. This isn’t something I want to look at right now, but I’ve moved it over to the discussions, and will have a think about it.
